core/src/components/datetime/datetime.tsx

```diff
@@ -1444,18 +1452,33 @@ export class Datetime implements ComponentInterface {
             <slot name="buttons">
               <ion-buttons>
                 {showDefaultButtons && (
-                  <ion-button id="cancel-button" color={this.color} onClick={() => this.cancel(true)}>
+                  <ion-button
+                    id="cancel-button"
+                    color={this.color}
+                    onClick={() => this.cancel(true)}
+                    disabled={this.disabled || this.readonly}
```

Comment:
I think it will be easier for other developers to understand this code if we move this check to its own variable. The reason is that we use it 3 times in this function, but it's not immediately clear why we check both disabled and readonly.

You could do something like this:

```js
const isButtonDisabled = this.disabled || this.readonly;

...

disabled={isButtonDisabled}
```

From there we'd also want comments above `isButtonDisabled` as to why we need to check disabled and readonly. Feel free to choose a different variable name.